helix-term/src/main.rs

```diff
@@ -109,7 +109,11 @@ FLAGS:
 
     // TODO: use the thread local executor to spawn the application task separately from the work pool
     let mut app = Application::new(args, config).context("unable to create new application")?;
-    app.run().await.unwrap();
+
+    let ret = app.run().await.unwrap();
+    if ret != 0 {
```

Comment:
Maybe there's a better way to do this, since "no destructors on the current stack or any other thread’s stack will be run". Maybe the easiest would be to make a "stub" main that will call this and change the return signature to `Result<i32>`? I don't know how that will play with tokio's `main` proc macro, though.

Reply:
Are you able to return a `Result<i32>` here with `tokio::main` or does it fail to compile? If it fails to work we can drop the main macro and just manually construct the runtime: https://docs.rs/tokio/1.13.0/tokio/runtime/struct.Builder.html

Reply:
Actually, looks like this works fine. Just pushed it, and everything seems to work fine. I'd be glad to squash if it looks fine to you (or let you squash-merge it, either-or).